front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/StorageServiceSupport.java

```diff
@@ -78,6 +78,24 @@ public StorageServiceSupport(ObjectType objectType,
     this.shouldWarmCache = shouldWarmCache;
     this.registry = registry;
 
+    // cache object keys for at least two refresh intervals
+    // (will ensure that concurrent requests are debounced across a refresh cycle)
+    long cacheExpiry = refreshIntervalMs * 2;
+    log.debug("Creating object keys cache (expiry: {} minutes)", TimeUnit.MILLISECONDS.toMinutes(cacheExpiry));
+    this.objectKeysByLastModifiedCache = CacheBuilder
+      .newBuilder()
```

Comment:
I'm a bit worried about the potential memory usage here if there are a bunch of cache loads and it takes 2x interval for things to expire - wonder about bounding the cache with a lru policy so older entries drop out and/or tracking previous keys and calling evict after a newer lastmodified loads successfully?

Reply:
I guess with just map<String,Long> it's not ever going to be too huge

Reply:
Thought briefly about the memory implications but figured `String -> Long` would be reasonably small, and 2 x interval is 120s. 

_Should_ be alright but I'll watch what happens with EntityTags ... could consider excluding them.